File: ga/4omini_result_2024_7/result/macro_func_mini/49_macro_func.cpp
use std::io;

fn cppmax(x: i32, y: i32) -> i32 {
    if x + y < 5 {
        y
    } else {
        x
    }
}

fn main() {
    let mut input_a = String::new();
    let mut input_b = String::new();
    
    println!("Enter two integers (a and b):");
    
    // Read first integer
    io::stdin().read_line(&mut input_a).expect("Failed to read input");
    let a: i32 = input_a.trim().parse().expect("Please enter a valid integer");
    
    // Read second integer
    io::stdin().read_line(&mut input_b).expect("Failed to read input");
    let b: i32 = input_b.trim().parse().expect("Please enter a valid integer");
    
    let result = cppmax(a, b);
    println!("Result: {}", result);
}

File: ga/glm4_result_2024_6/result/macro_func_glm/49_macro_func.rs
use std::io;

fn main() {
    let mut input = String::new();

    println!("Enter two numbers separated by space: ");
    io::stdin().read_line(&mut input).expect("Failed to read line");

    let numbers: Vec<i32> = input.trim().split_whitespace()
        .map(|s| s.parse().expect("Please enter a valid number"))
        .collect();

    let (a, b) = (numbers[0], numbers[1]);

    let result = cppmax(a, b);
    println!("Result: {}", result);
}

fn cppmax(x: i32, y: i32) -> i32 {
    if x + y < 5 {
        y
    } else {
        x
    }
}